flytekit/remote/remote.py

```diff
@@ -524,7 +524,9 @@ def get_launch_plan_from_then_node(
 
                 if node.branch_node:
                     get_launch_plan_from_branch(node.branch_node, node_launch_plans)
-        return FlyteWorkflow.promote_from_closure(compiled_wf, node_launch_plans)
+        flyte_workflow = FlyteWorkflow.promote_from_closure(compiled_wf, node_launch_plans)
+        flyte_workflow.template._id = workflow_id
+        return flyte_workflow
```

Comment:
This bit isn't necessary, but it looks like tasks already do this and it makes it easier to unittest.